Assets/Scripts/EnemyLogic/Movement_Pursuit.cs

```diff
@@ -0,0 +1,106 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+/* Author:  Fouche', Els
+ * Updated: 04/28/2024
+ * Notes:   Enemy logic for pursuing the player when spotting them.
+ */
+
+public class Movement_Pursuit : MonoBehaviour
+{
+    [Header("Pursue Player")]
+    public int checkForPlayer = 5;
+    public float timeBetweenChecks = 1.0f;
+
+    private GameObject target;
+    private NavMeshAgent navMeshAgent;
+    private EnemyController enemyController;
```

Comment:
This is a cyclical dependency, I'd be cautious about this. If they rely on eachother so much, they might be better off beign the same class, if its just for few function calls I saw, then maybe it could be better to restructure so there isn't this dependency.

Reply:
Yeah this might be overly-decomposed.